logging/logging.go

```diff
@@ -43,3 +46,21 @@ func (f *textFormatter) Format(entry *log.Entry) ([]byte, error) {
 	b.WriteByte('\n')
 	return b.Bytes(), nil
 }
+
+// WithContext returns a logger that has information from the context as fields.
+//
+// e.g.
+//     logger := logging.WithContext(ctx)
+//     logger.Errorf("Some error")
+func WithContext(ctx context.Context) *log.Entry {
```

Comment:
How do they chain?

TBH I'd prefer if these lived in their various packages - ie this was in `users` as `Logger`, and there was one in the tracing package etc.  These could then be combined in a standard weaveworks logger here.  But just a thought.